src/kOS.Safe/Exceptions/KOSInvalidArgumentException.cs

```diff
@@ -0,0 +1,11 @@
+namespace kOS.Safe.Exceptions
+{
+    public class KOSInvalidArgumentException : KOSException
+    {
+        private const string MSG = "While Invoking function {0}, argument {1} was invalid for the reason: {2}";
+
+        public KOSInvalidArgumentException(string functionName, string argumentName, string reason) : base(string.Format(MSG, functionName, argumentName, reason))
+        {
+        }
+    }
+}
```

Comment:
We probably need to go back and find other places where this can be thrown too.  It looks like it could be used all over the place.